### DISCUSSION TITLE
Thanks to Pyodide, marimo, our open-source reactive notebook for Python now runs in the browser

### DISCUSSION THREAD

**Discussion by akshayka** [pyodide/pyodide#4594] [Show and tell]
Hi Pyodide community! 👋

I wanted to say thank you for creating such a powerful and magical product. I'm one of the developers of [marimo](https://github.com/marimo-team/marimo), an open-source reactive notebook for Python. Thanks to Pyodide, marimo now [runs entirely in the browser](https://marimo.app/l/c7h6pz)!

We were pleasantly surprised how simple it was for us to adapt our codebase to work via Pyodide, how fast Pyodide is, and how many packages for scientific computing are already available. We're very excited by the potential of our [Pyodide-powered notebook](https://docs.marimo.io/guides/wasm.html) and others like it: it makes it very easy for students to get up and running with Python, makes it possible to publish interactive blog posts, documentations, and tools as part of static web pages, and generally gives anyone instant access to a Python canvas for experimenting with code and data.

We have many more features we plan to support, including better integration with auxiliary files and data. It seems like the tools that Pyodide provides will go a long way toward us implementing these features. For example, we're currently making use of the relatively new `SharedArrayBuffer`-based interrupt handler to make notebook execution preemptible.

Thanks so much for the incredible project. I think it can't be overstated the extent to which your project will make Python-based computing more accessible for everyone in the coming years. Really excited for Pyodide's future.

I noticed the Pyodide team has a blog. If you're willing, I'd love to write a guest post about how and why we adapted our notebook to be runnable in the browser via Pyodide. Please let me know if that's something that would be of interest.

**Comment by ryanking13** [comment]
Thank you for sharing your experience with Pyodide! Feedback from the community is very encouraging and helpful to us.

> I noticed the Pyodide team has a blog. If you're willing, I'd love to write a guest post about how and why we adapted our notebook to be runnable in the browser via Pyodide. Please let me know if that's something that would be of interest.

Sure. We'd love it if you could write a blog post about how you've applied Pyodide to marimo.

**Reply by akshayka** [reply]
You're welcome! It's really a fantastic project.

> Sure. We'd love it if you could write a blog post about how you've applied Pyodide to marimo.

That's great! I'll start on something. I can share it here with a Google Doc link when I have a draft, if that works?

**Reply by hoodmane** [reply]
Usually what we do is:
1. Open an issue on https://github.com/pyodide/pyodide-blog with an outline so that we can agree on the general idea before you spend too much time fleshing everything out. We generally approve these.
2. Once we agree on the outline, write the post in markdown as a PR against pyodide/pyodide-blog

We are willing to accept slightly different workflows if you have another preference, for instance sharing a google doc link would be fine. But we do want the final product in markdown as a PR.

**Reply by akshayka** [reply]
Your suggested workflow is perfect; thanks for taking the time to explain it. I see now the instructions are at your repo — sorry I missed that!
